buildSrc/src/main/java/com/google/firebase/gradle/plugins/FirebaseJavaLibraryPlugin.kt

```diff
@@ -0,0 +1,101 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.gradle.plugins
+
+import com.github.sherter.googlejavaformatgradleplugin.GoogleJavaFormatExtension
+import com.google.common.collect.ImmutableList
+import com.google.common.collect.ImmutableMap
+import java.io.File
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.attributes.Attribute
+import org.gradle.api.plugins.JavaPluginConvention
+import org.gradle.api.publish.PublishingExtension
+import org.gradle.api.publish.maven.MavenPublication
+import org.gradle.kotlin.dsl.getPlugin
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+
+class FirebaseJavaLibraryPlugin : Plugin<Project> {
+
+  override fun apply(project: Project) {
+    project.apply(ImmutableMap.of("plugin", "java-library"))
```

Comment:
I added a little util method to your PR, you should be able to replace these with something more Kotlin like:
```kotlin
project.pluginManager.apply<LibraryPlugin>()
project.pluginManager.apply<LicenseResolverPlugin>()
project.pluginManager.apply<GoogleJavaFormatPlugin>()
```